🚨 Vercel Deploy failed on dev

**Workflow**: Vercel Deploy
**Branch**: dev
**Commit**: 6cc783caa093ff50fc703d24899cdf4737abca5c
**Run**: https://github.com/Mark-Yun/minglit/actions/runs/23579466512
**Triggered by**: N/A
**Actor**: Mark-Yun

🤖 TPM 근본 원인 분석 (2026-03-27)

**실패율**: 최근 7일 Deploy to Vercel 11/11 실패 (100%)

**Root cause**: `landing_partner` 빌드 시 `env-manifest.json` 모듈을 찾을 수 없음

```
Error: Cannot find module '../../../env-manifest.json'
Require stack:
- /vercel/path0/lib/env.ts
- /vercel/path0/next.config.compiled.js
```

**원인**: `lib/env.ts:11`에서 `require('../../../env-manifest.json')` 호출하나, Vercel 빌드 환경에서 해당 경로에 파일이 존재하지 않음. 모노레포 root의 `env-manifest.json`이 Vercel의 root directory 설정(`apps/landing_partner`) 밖에 위치하기 때문으로 추정.

**수정 방향**:
1. `env-manifest.json`을 Vercel 빌드 경로에서 접근 가능하도록 배치 (install command에서 복사 또는 symlink)
2. 또는 `lib/env.ts`의 import 경로를 Vercel 환경에 맞게 수정
3. 또는 Vercel project 설정에서 Root Directory를 모노레포 root로 변경

--- a/apps/landing_partner/lib/env.ts
+++ b/apps/landing_partner/lib/env.ts
@@ -1,11 +1,12 @@
-import manifest from "../../../env-manifest.json";
-
 /**
  * Validate that all required build-time env vars for landing_partner are present.
  *
  * Call from `next.config.ts` so missing variables fail the build early.
  */
 export function validateBuildEnv(): void {
+  // Fix #447: lazy require to avoid module resolution failure on Vercel
+  // where the monorepo root is outside the build directory
+  const manifest = require("../../../env-manifest.json");
   const section = manifest.nextjs.landing_partner;
   const required = Object.keys(section.required);
   const missing = required.filter((k) => !process.env[k]);

--- a/apps/landing_partner/lib/env.ts
+++ b/apps/landing_partner/lib/env.ts
@@ -6,6 +6,7 @@
 export function validateBuildEnv(): void {
   // Fix #447: lazy require to avoid module resolution failure on Vercel
   // where the monorepo root is outside the build directory
+  // eslint-disable-next-line @typescript-eslint/no-require-imports
   const manifest = require("../../../env-manifest.json");
   const section = manifest.nextjs.landing_partner;
   const required = Object.keys(section.required);

--- a/apps/landing_user/lib/env.ts
+++ b/apps/landing_user/lib/env.ts
@@ -1,11 +1,12 @@
-import manifest from "../../../env-manifest.json";
-
 /**
  * Validate that all required build-time env vars for landing_user are present.
  *
  * Call from `next.config.ts` so missing variables fail the build early.
  */
 export function validateBuildEnv(): void {
+  // Fix #447: lazy require to avoid module resolution failure on Vercel
+  // where the monorepo root is outside the build directory
+  const manifest = require("../../../env-manifest.json");
   const section = manifest.nextjs.landing_user;
   const required = Object.keys(section.required);
   const missing = required.filter((k) => !process.env[k]);

--- a/apps/landing_user/lib/env.ts
+++ b/apps/landing_user/lib/env.ts
@@ -6,6 +6,7 @@
 export function validateBuildEnv(): void {
   // Fix #447: lazy require to avoid module resolution failure on Vercel
   // where the monorepo root is outside the build directory
+  // eslint-disable-next-line @typescript-eslint/no-require-imports
   const manifest = require("../../../env-manifest.json");
   const section = manifest.nextjs.landing_user;
   const required = Object.keys(section.required);

